hoss/spatial.py

```diff
@@ -116,6 +122,22 @@ def get_spatial_index_ranges(
                     )
                 )
 
+        if (not geographic_dimensions) and (not projected_dimensions):
```

Comment:
I suspect this still isn't the right condition to check for things. There's a previous comment (one of the many, sorry!) about it being better to instead check if there are any required variables with anonymous dimensions (see [here](https://github.com/nasa/harmony-opendap-subsetter/pull/15#discussion_r1769244311)). I think that's really what you want to check, instead of if there are geographic or projected spatial dimensions.

Reply:
if there are geographic_dimensions or projected_dimensions , we don't want to get coordinate variables right?

Reply:
Capturing some Slack conversation:

If you did have a function to get a list/set of variables that have anonymous dimensions, maybe the cleanest thing is something more like:

```
variables_with_anonymous_dims = get_variables_with_anonymous_dims(...)

for variable in variables_with_anonymous_dims:
    latitude_coordinates, longitude_coordinates = get_coordinate_variables(varinfo, variable)
    if latitude_coordinates and longitude_coordinates:
        index_ranges.update(
            get_x_y_index_ranges_from_coordinates(...)
        )
```

Also - noting that I think the latitude_coordinates and longitude_coordinates should only be the coordinates for that specific variable.

(If the return value from `get_variables_with_anonymous_dims(...)` is an empty list, then the `for-in` loop becomes a no-op because there's nothing to iterate over.)

Reply:
Also, I really like the name of `get_x_y_index_ranges_from_coordinates` - I think it helps with scanning the code and understanding what's going on here.

Reply:
updated that method - https://github.com/nasa/harmony-opendap-subsetter/pull/15/commits/60fb22a098e173c77b50fbb043045e56ad06f877

Reply:
Thanks for the update. I'll mark this conversation as resolved, and review the new version.